hbase-client/src/main/java/org/apache/hadoop/hbase/zookeeper/HQuorumPeer.java

```diff
@@ -85,14 +85,20 @@ public static void main(String[] args) {
   }
 
   private static void runZKServer(QuorumPeerConfig zkConfig) throws UnknownHostException, IOException {
```

Comment:
The current signature proposes we throw IOException or InterruptedException. ZK 3.6 throws another type of checked exception, which causes a compilation error. Therefore I catch that and potentially others, and wrap it into an IOE.

Adding the new checked exception to the signature would cause a compilation problem with 3.4, so that is not possible due to this legacy.

Reply:
> UnknownHostException is also an IOE

Might be true but the method signature here was what it was and there is no need to fix it. I have little opinion either way other than a preference to minimize change.